03week/ticTacToe.js

```diff
@@ -23,24 +36,86 @@ function printBoard() {
   console.log('2 ' + board[2].join(' | '));
 }
 
-function horizontalWin() {
-  // Your code here
+const isValidPick = (row,column) =>{
+  return board[row][column] == ' ';
+}
+
+const isEntryValid = (row,column) =>{
+  return row >= 0 && row <= 2 && column >= 0 && column <= 2;
+}
+
+const switchPlayers = () =>{
+  if (playerTurn == 'X'){
+    playerTurn = 'O'
+  }else{
+    playerTurn = 'X';
+  }
+}
+
+//try array.every() for horiz win once get working properly
+const horizontalWin = () => {
+  for(let x=0;x<3;x++){
+    if(board[x][0] == board[x][1] && board[x][1] == board[x][2] && board[x][1] != ' '){
+      isItAWinner = true;
+      console.log('horiz win')
+    }
+  }
+  return isItAWinner;
 }
 
-function verticalWin() {
-  // Your code here
+const verticalWin = () => {
+  for(let y=0;y<3;y++){
+    if(board[0][y] == board[1][y] && board[1][y] == board[2][y] && board[1][y] != ' '){
+      isItAWinner = true;
+      console.log('vert win')
+    }
+  }
+  return isItAWinner;
 }
 
-function diagonalWin() {
-  // Your code here
+const diagonalWin = () => {
+  if(board[0][0] == board[1][1] && board[1][1] == board[2][2] && board[1][1] != ' ' || board[0][2] == board[1][1] && board[1][1] == board[2][0] && board[1][1] != ' '){
+    isItAWinner = true;
+    console.log('diag win');
+  }
+  return isItAWinner;
 }
 
-function checkForWin() {
-  // Your code here
+const checkForWin = () => {
+  return horizontalWin() || verticalWin() || diagonalWin()
 }
 
-function ticTacToe(row, column) {
-  // Your code here
+const checkForDraw = ()=>{
+  return !board.every((boardArray)=>{
+    return boardArray.every((IndividualArray)=>{
+      return IndividualArray == ' ';
+    });
+  });
+}
+
+const resetBoard = () =>{
+  board.forEach((x,index) => {
+    board[index].fill(' ');
+  });
+  //when running through the foreach, and not using the element value, still need to put in a placeholder, x, in order to access the index, right?
+  playerTurn = 'X';
+  console.log("C'mon! Let's play again!");
+}
+
+const ticTacToe = (row,column) => {
+  if (isEntryValid(row,column) && isValidPick(row,column)){
```

Comment:
The reason that your pieces are not able to get on the board is because you IF statement is not passing.
if statement only want to the know if the evaluation is a Boolean